PropTechAPI - コピー/src/main/java/com/PropTech/UserRegist/PropTechAppAPI/PropTechAPI/Logic/UserRegistrationLogic.java

```diff
@@ -0,0 +1,213 @@
+			//System.out.println(exception);
+			return GenerateResponse(ResponseCommon.E500);
+		}finally {
+			//System.out.println("完了");
+		}
+		return Response.status(Response.Status.CREATED.getStatusCode()).entity(responsedetails).build();
+	}
+	
+	private Response GeneratedHashPassword(RequestBody body, StringBuilder HashPassword) {
+		ResponseDetails responsedetails = new ResponseDetails();
+		String SHA256Password = PasswordHashUtil.getSaftyPassword("password", body.getpassword());
+		HashPassword.append(SHA256Password);
+		return Response.status(Response.Status.CREATED.getStatusCode()).entity(responsedetails).build();
+	}
+	
+	private Response GeneratedbConnectionSelect(StringBuilder HashPassword) {
+		ResponseDetails responsedetails = new ResponseDetails();
+		try(Connection connection = DriverManager.getConnection(URL, USER, PASSWORD);
+				PreparedStatement preparedStatement = connection.prepareStatement("SELECT * FROM userregistrationapi WHERE password = ?")){
+			preparedStatement.setString(1, HashPassword.toString());
+			
+			try (ResultSet resultset = preparedStatement.executeQuery()){
+				while(resultset.next()) {
+					System.out.println(resultset.getString("password"));
+					if(resultset.getString("password").equals(null)) {
+						//Not Execute
+					}else {
+						return GenerateResponse(ResponseCommon.E500);
+					}
+				}
+			}catch(SQLException sqlexception) {
+				sqlexception.printStackTrace();
+				return GenerateResponse(ResponseCommon.E500);
+			}
+		}catch(SQLException exception) {
+			exception.printStackTrace();
+			return GenerateResponse(ResponseCommon.E500);
+		}finally {
+
+		}
+		return Response.status(Response.Status.CREATED.getStatusCode()).entity(responsedetails).build();
+	}
+	
+	private Response GenerateddbConnectionInsert(RequestBody body, StringBuilder seqID, StringBuilder HashPassword) {
+		ResponseDetails responsedetails = new ResponseDetails();
+		
+		int age = Integer.parseInt(body.getage());
+		//System.out.println(age);
+		
+		try(Connection connection = DriverManager.getConnection(URL, USER, PASSWORD)){
+			connection.setAutoCommit(false);
+			try(PreparedStatement preparedStatement = connection.prepareStatement("INSERT INTO userregistrationapi("
+						+ "api_seq_no,"
+						+ "sex,"
+						+ "age,"
+						+ "profession,"
+						+ "username,"
+						+ "password,"
+						+ "token)"
+						+ "VALUES(?,?,?,?,?,?,?)")){
+				preparedStatement.setString(1, seqID.toString());
+				preparedStatement.setString(2, body.getsex());
+				preparedStatement.setInt(3, age);
+				preparedStatement.setString(4,  body.getprofession());
+				preparedStatement.setString(5,  body.getusername());
+				preparedStatement.setString(6,  HashPassword.toString());
+				preparedStatement.setString(7,  body.gettoken());
+				preparedStatement.executeUpdate();
+				connection.commit();
+				
+			}catch(Exception exception) {
+				connection.rollback();
+				return GenerateResponse(ResponseCommon.E500);

```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the resource returned by the following method call: *GenerateResponse*. The resource is returned at line 199. There are other execution paths that don&#39;t close the resource or return it, for example, when PreparedStatement.setInt throws an exception. Ensure the object returned by *GenerateResponse()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).